### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by bvego**
Added facebook and google login on the /login route

### CODE REVIEW

File: src/config/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export const BASE_URL = "http://localhost:5000";
```
Comment (wr9) [inline src/config/index.js]: I would maybe put this is a API urls constants file, where we would later add other needed subroutes/url parts

File: src/reducers/auth.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import api from 'utils/api';
+
+const LOGIN_START = 'rmp/users/LOGIN_START';
+const LOGIN_SUCCESS = 'rmp/users/LOGIN_SUCCESS';
+const LOGIN_FAILURE = 'rmp/users/LOGIN_FAILURE';
+
+const initialState = {
+  isLoggedIn: false,
+  isLoggingIn: false,
```
Comment (wr9) [inline src/reducers/auth.js]: maybe better to name this "loading" so it's consistent across other states we'll have?

File: src/reducers/auth.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import api from 'utils/api';
+
+const LOGIN_START = 'rmp/users/LOGIN_START';
```
Comment (wr9) [inline src/reducers/auth.js]: I think this should be rmp/auth/LOGIN

File: src/reducers/auth.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const LOGIN_SUCCESS = 'rmp/users/LOGIN_SUCCESS';
+const LOGIN_FAILURE = 'rmp/users/LOGIN_FAILURE';
+
+const initialState = {
+  isLoggedIn: false,
+  isLoggingIn: false,
+  error: '',
```
Comment (wr9) [inline src/reducers/auth.js]: would prefer null

File: src/reducers/auth.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export const loginUser = (service, userData) => async dispatch => {
+  dispatch({ type: LOGIN_START });
+  try {
+    localStorage.setItem('token', userData.accessToken);
+    localStorage.setItem('service', service);
```
Comment (wr9) [inline src/reducers/auth.js]: just a suggestion - we can use indexed db, so its not blocking and we can simply use promises?

File: src/routes/LandingPage/components/AuthButton/index.jsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { cloneElement } from "react";
+
+const AuthButton = props =>
+  cloneElement(props.children, {
```
Comment (wr9) [inline src/routes/LandingPage/components/AuthButton/index.jsx]: do we need cloneElement here?

File: package.json [hunk 7]
```diff
@@ -4,8 +4,9 @@
   "private": true,
   "dependencies": {
     "autoprefixer": "8.0.0",
+    "axios": "^0.18.0",
     "babel-core": "6.26.0",
-    "babel-eslint": "8.2.2",
+    "babel-eslint": "^7.2.3",
```
Comment (ikovic) [inline package.json]: Was downgrade done on purpose?

File: src/reducers/auth.js (hunk 2 above)
Comment (ikovic) [reply src/reducers/auth.js]: That makes sense while you have only a single `loading` status in a redux module.
If there is more than one, meaningful names help a lot.
I'd go with something like `logInInProgress`.

File: src/reducers/auth.js (hunk 4 above)
Comment (ikovic) [reply src/reducers/auth.js]: Error will probably be an object so empty string might be misleading here.

Comment (ikovic) [reply src/reducers/auth.js]: Even though `null === false && '' === false`

File: src/reducers/auth.js (hunk 5 above)
Comment (ikovic) [reply src/reducers/auth.js]: Local storage is fine for this purpose.

File: src/routes/LandingPage/components/AuthButton/index.jsx (hunk 6 above)
Comment (ikovic) [reply src/routes/LandingPage/components/AuthButton/index.jsx]: It's probably used to pass props to children.
Maybe a cleaner approach would be to use render props?
https://reactjs.org/docs/render-props.html

File: src/routes/LandingPage/components/FacebookLogin/index.jsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import FacebookLoginComponent from "react-facebook-login";
+import { FACEBOOK_ID } from "config/auth";
+
+const FacebookLogin = ({ onSuccess, onFailure, className }) => {
+  return (
```
Comment (ikovic) [inline src/routes/LandingPage/components/FacebookLogin/index.jsx]: No need for explicit return.

File: src/routes/LandingPage/components/FacebookLogin/index.jsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    <FacebookLoginComponent
```
Comment (ikovic) [inline src/routes/LandingPage/components/FacebookLogin/index.jsx]: No need for `Component` in name

File: src/routes/LandingPage/components/GoogleLogin/index.jsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import GoogleLoginComponent from "react-google-login";
```
Comment (ikovic) [inline src/routes/LandingPage/components/GoogleLogin/index.jsx]: No need for `Component` in name

File: src/utils/api.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    config.headers.Service = activeService;
+
+    return config;
+  },
+  error => {
+    console.log('interceptor error', error);
+  },
+);
+
+export const loginUser = () => instance.post(`${BASE_URL}/api/login`);
```
Comment (ikovic) [inline src/utils/api.js]: Shouldn't this be in according reducer?

**Comment by wr9** [comment]
👍

File: src/routes/LandingPage/components/AuthButton/index.jsx [hunk 12]
```diff
@@ -1,8 +1,10 @@
-import { cloneElement } from "react";
+import React from 'react';
 
-const AuthButton = props =>
-  cloneElement(props.children, {
-    className: "h3 f6 dim ph3 pv2 mb5 dib bg-lightest-blue pointer black"
-  });
+const AuthButton = props => (
+  <props.component
```
Comment (ikovic) [inline src/routes/LandingPage/components/AuthButton/index.jsx]: Does not look pretty but works for this use case.

**Review by ikovic — APPROVED** [review]

**Pull request merged**